src/riot-rs-threads/src/lib.rs

```diff
@@ -276,6 +473,28 @@ impl Threads {
 /// Currently it expects at least:
 /// - Cortex-M: to be called from the reset handler while MSP is active
 pub unsafe fn start_threading() {
+    #[cfg(feature = "multicore")]
+    {
+        // Idle thread that prompts the core to enter deep sleep.
+        fn idle_thread() {
+            loop {
+                Cpu::wfi();
+            }
+        }
+
+        // Stacks for the idle threads.
+        // Creating them inside the below for-loop is not possible because it would result in
+        // duplicate identifiers for the created `static`.
+        static STACKS: [ConstStaticCell<[u8; IDLE_THREAD_STACK_SIZE]>; CORES_NUMOF] =
+            [const { ConstStaticCell::new([0u8; IDLE_THREAD_STACK_SIZE]) }; CORES_NUMOF];
+
+        // Create one idle thread for each core with lowest priority.
+        for stack in &STACKS {
+            thread_create_noarg(idle_thread, stack.take(), 0, None);
```

Comment:
Core affinities introduce some overhead (and are thus also feature-gated), and there is not need for them here.
As long as we have as many idle threads as cores, it doesn't matter which core one executes which thread. 

Or is there a case that I am missing, where not having the idle threads pinned to a core would lead to a problem?

Reply:
I don't see where this might cause a problem! Was just wondering, avoiding overhead makes sense of course.